### PR TITLE
Shielded suit rework

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Reworks how energy shield functions.
- Energy shields now work off integrity instead of charges. Higher damage weapons will break shields faster, and no damage weapons like the disabler will not damage shilelds.
- Shields can set what they protect against.
- Cultist armour will no longer protect against melee attacks, but will protect from 100 projectile damage before permanently breaking.
- ~~Shields now flash when hit.~~ NVM Animations don't work on overlays properly

For the most part, 1 charge = 20 damage. The only exception is cultists, who were given 100 health on their shield instead of 60, with the compensatino being that it doesn't protect against melee attacks.

This is largely a reversion of the shield component update seen in #10992 as this completely replaces all behaviours of the shield component add in that PR.

## Why It's Good For The Game

The charge system on shields is pretty lame and doesn't account for weak or powerful weapons damaging them. This means that you can cheese stronger armours with weapons that do nothing but fire fast, like the temperature gun.

In addition to this, more flexability is given to shields. For example, cult shields will no longer block melee attacks but will protect against ranged attacks.

## Testing Photographs and Procedure

image

## Changelog
:cl:
balance: Reworks energy shields to be based on integrity rather than charges.
balance: Shielded syndicate hardsuits will now protect against 60 total damage rather than 3 attacks. It regenerates 10 health per second after 20 seconds of not being attacked.
balance: The battlemage suit will grant 300 damage protection, without the ability to recharge the shield over time. Each battlemage shield charge will provide 400 shield health.
balance: Cult shielded hardsuits will protect against 100 damage before permanently breaking, but will not defend against melee attacks.
/:cl:

### CODE REVIEW

**Comment by Howluinb** [comment]
If you're reworking on shielded suits, how will you tackle the wizard one ? 
It severely needs a buff, because a wizard planning to last more than 20 seconds would never take a piece of armor that never recharges

**Comment by BlueHNT** [comment]
I ain't gonna sugarcoat it. Not being able to shoot a horde of cultists gives them a free win since they 3 hit crit. Having 400 shield power as cultist is ridiculous and frankly unfair. Especially since they rush down in groups.

File: code/modules/clothing/spacesuits/hardsuit.dm [hunk 1]
```diff
@@ -914,9 +914,23 @@
...
+	var/shield_icon = "shield-old"
+
+/obj/item/clothing/suit/space/hardsuit/shielded/Initialize(mapload)
+	. = ..()
+	if(!allowed)
+		allowed = GLOB.advanced_hardsuit_allowed
 
-/obj/item/clothing/suit/space/hardsuit/shielded/setup_shielding()
-	AddComponent(/datum/component/shielded, max_charges = 3, recharge_start_delay = 20 SECONDS, charge_increment_delay = 1 SECONDS, charge_recovery = 1, lose_multiple_charges = FALSE, shield_icon = "shield-old")
+/obj/item/clothing/suit/space/hardsuit/shielded/ComponentInitialize()
```
Comment (Tsar-Salat) [inline code/modules/clothing/spacesuits/hardsuit.dm]: any reason to use componentinit over `setup_shielding` wrapper?

Comment (PowerfulBacon) [reply code/modules/clothing/spacesuits/hardsuit.dm]: oh, no

**Pull request merged**